### PR TITLE
Respect files without styles

### PR DESCRIPTION
**Pull request by velios**
@kbosompem Hi. I started making a fix for the case where parsing file with missing "r" cell attributes. Like it [described here](https://github.com/box/spout/issues/267) for project like bb-excel but for PHP. But i can't finish it without some refactoring. Therefore, I first decided to fix a small problem where the style file was missing and did a small refactoring in it. At the same time, I’ll find out how you feel about refactoring. If you upload PR, please use squash&merge. Have a good day.

### CODE REVIEW

File: src/bb_excel/core.clj [hunk 1]
```diff
@@ -132,68 +133,74 @@
...
+        cell* (merge cell
+                     {:x row-index
+                      :y col-index})
+        cell-type (:t cell*)
+        cell-value (:d cell*)]
     (cond
-      ;; Possible data types well explained here https://stackoverflow.com/a/18346273
-      (= (:t u) "s")                (dissoc (assoc-in u [:d] (dict (read-string (:d u)))) :t)
-      (= (:t u) "str")              (dissoc u :t)
-      (= (:t u) "inlineStr")        (dissoc u :t)
```
Comment (velios) [inline src/bb_excel/core.clj:144]: don't understand why we need dissoc :t, we completely transform this data in next step

File: src/bb_excel/core.clj [hunk 2]
```diff
@@ -132,68 +133,74 @@
...
+  [row]
+  (parse-long (:y row)))
 
 (defn process-row
   "Process Excel row of data"
-  [dict styles coll]
-  (reduce #(merge % {:_r (read-string (:y %2)) (keyword (:x %2)) (:d %2)}) {}
-          (map (partial process-cell dict styles)
-               (map #(merge (first %) {:d (second %)}
-                            {:f (nth % 2)})
```
Comment (velios) [inline src/bb_excel/core.clj:159]: Don't see any usage of :f on next steps, so i wipe it

File: src/bb_excel/core.clj [hunk 3]
```diff
@@ -132,68 +133,74 @@
...
-                    (map (juxt :attrs
-                               (comp last :content last :content)
-                               (comp first :content first :content)) coll)))))
+  [dict styles row]
+  (let [row* (->> row
+                  (map (fn [cell] (merge (:attrs cell)
+                                         {:d (-> cell :content last :content last)})))
+                  (map (partial process-cell dict styles)))
+        row-index (get-row-index (first row*))]
+    (into {:_r row-index} (map #(-> [(keyword (:x %)) (:d %)]))
```
Comment (velios) [inline src/bb_excel/core.clj:173]: #(-> <some new data form>) maybe not clear syntax, but it's just response with new data form

File: src/bb_excel/core.clj [hunk 4]
```diff
@@ -274,7 +281,7 @@
 (defn parse-range
   "Takes in an Excel coordinate and returns a hashmap of rows and columns to pull"
   [s]
-  (let [[[_ osc osr oec oer]] (re-seq #"([A-Z]+)([0-9]*)[:]?([A-Z]*)([0-9]*)" s)
+  (let [[_ osc osr oec oer] (re-matches #"([A-Z]+)([0-9]*)[:]?([A-Z]*)([0-9]*)" s)
```
Comment (velios) [inline src/bb_excel/core.clj:288]: so we can destructure without an extra layer of brackets

File: src/bb_excel/core.clj [hunk 5]
```diff
@@ -132,68 +133,74 @@
...
+      (->> (xml-seq x)
+           (filter #((:text-part tags) (:tag %)))
+           (map get-cell-text)
+           (zipmap (range))))
     {}))
 
 (defn get-styles
   "Get styles"
   [^ZipFile zipfile]
-  (let [wb  (.getEntry zipfile (str "xl/styles.xml"))
```
Comment (velios) [inline src/bb_excel/core.clj:187]: Don't understand why we need str here

**Review by velios — COMMENTED** [review]
Some explanation comments

**Review by velios — COMMENTED** [review]
1

File: src/bb_excel/core.clj [hunk 6]
```diff
@@ -67,6 +68,11 @@
                  :xmlns.http%3A%2F%2Fschemas.openxmlformats.org%2FofficeDocument%2F2006%2Frelationships/id
                  :xmlns.http%3A%2F%2Fpurl.oclc.org%2Fooxml%2FofficeDocument%2Frelationships/id}})
 
+(def ^:const SHEET_TAG_TAGS (:sheet-tag tags))
+(def ^:const TEXT_PART_TAGS (:text-part tags))
+(def ^:const TEXT_T_TAGS (:text-t tags))
+(def ^:const SHEET_DATA_TAGS (:sheet-data tags))
```
Comment (velios) [inline src/bb_excel/core.clj:74]: Performance difference on huge data sets

File: src/bb_excel/core.clj [hunk 7]
```diff
@@ -132,68 +138,73 @@
...
     (let [ins (.getInputStream zipfile wb)
           x (parse-str (slurp ins))]
-      (->>
-       (filter #((:text-part tags) (:tag %)) (xml-seq x))
-       (map get-cell-text)
-       (zipmap (range))))
-    {}))
+      (into [] (comp (filter #(contains? TEXT_PART_TAGS (:tag %)))
+                     (map get-cell-text))
+            (xml-seq x)))
```
Comment (velios) [inline src/bb_excel/core.clj:192]: Actually we don't need map with indexes. Vector make access by index in O(1) time with nth. Increase perfomance 20% on my sample.

**Review by velios — COMMENTED** [review]
Some perfomance improvements

**Comment by velios** [comment]
@kbosompem Hello. Can I hope that you will look at my PR in the near future?

File: src/bb_excel/core.clj (hunk 1 above)
Comment (kbosompem) [reply src/bb_excel/core.clj:144]: Can't remember why either. Ok to remove.

**Pull request merged**